### DISCUSSION TITLE
Git Branch Switching + File Watching: Where is this done? Prior convos?

### DISCUSSION THREAD

**Discussion by tony** [webpack/webpack#13538] [General]
Hi,

When running webpack, `git checkout`'s / other VCS switching can be far more drastic than a normal file save:

- Multiple changes in JS
- Code generated pre-processed files changed
- package.json file changes
- webpack config changes

Is file watching part of webpack itself? Or is it do with things like webpack-dev-server?

Is there a hook specifically to detect vcs branch switches alongside file diffs? **If so, that'd a time to raise a hook as a priority that can potentially interrupt other things that'd be triggered**. The reason why is for a vcs branch change, the desired effect may be a full restart - following the normal watch options makes my computer freeze

How much of this is under webpack's jurisdiction?  Any conversations from this tracker? Or projects? Any best practices?

**Comment by alexander-akait** [comment]
Please create reproducible example

**Reply by tony** [reply]
My intention is / was to see if there were prior conversations I wasn't aware of. Sorry about that!

I don't have time to reproduce this as my projects are private and complex, and I have no idea how long or what it'd take to do so. Can I close this?

**Reply by alexander-akait** [reply]
If you change branch and :
```
package.json file changes
webpack config changes
```

was changes happens you need to rerun webpack, we are working on it
